pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -2,61 +2,75 @@ package clusterconfig
 
 import (
 	"context"
+	"encoding/base64"
+	"encoding/pem"
 	"fmt"
 	"regexp"
 	"sort"
 	"strings"
 	"sync"
 	"time"
 
-	configv1 "github.com/openshift/api/config/v1"
-	"github.com/openshift/client-go/config/clientset/versioned/scheme"
-	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
-	certificatesv1beta1 "k8s.io/client-go/kubernetes/typed/certificates/v1beta1"
-
-	"encoding/base64"
-	"encoding/pem"
-
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
 	"k8s.io/apimachinery/pkg/util/json"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/apimachinery/pkg/util/sets"
 	kubescheme "k8s.io/client-go/kubernetes/scheme"
+	certificatesv1beta1 "k8s.io/client-go/kubernetes/typed/certificates/v1beta1"
 	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
 	"k8s.io/client-go/rest"
 	"k8s.io/klog"
 
+	configv1 "github.com/openshift/api/config/v1"
+	registryv1 "github.com/openshift/api/imageregistry/v1"
+	openshiftscheme "github.com/openshift/client-go/config/clientset/versioned/scheme"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	imageregistryv1 "github.com/openshift/client-go/imageregistry/clientset/versioned/typed/imageregistry/v1"
+
 	"github.com/openshift/insights-operator/pkg/record"
 )
 
 var (
-	serializer     = scheme.Codecs.LegacyCodec(configv1.SchemeGroupVersion)
-	kubeSerializer = kubescheme.Codecs.LegacyCodec(corev1.SchemeGroupVersion)
+	openshiftSerializer = openshiftscheme.Codecs.LegacyCodec(configv1.SchemeGroupVersion)
+	kubeSerializer      = kubescheme.Codecs.LegacyCodec(corev1.SchemeGroupVersion)
 
 	// maxEventTimeInterval represents the "only keep events that are maximum 1h old"
 	// TODO: make this dynamic like the reporting window based on configured interval
 	maxEventTimeInterval = 1 * time.Hour
+
+	registrySerializer serializer.CodecFactory
+	registryScheme     = runtime.NewScheme()
 )
 
+func init() {
+	utilruntime.Must(registryv1.AddToScheme(registryScheme))
```

Comment:
Is is neccessary to register the registryv1 explicitly ? I would think it would be registered automatically by importing the package, like it is happening with openshift/kubernetes types. Or am I missing something ?

Reply:
it is always better to make your own local scheme than poisoning or depending on some "automatic" behavior...

Reply:
> Is is neccessary to register the registryv1 explicitly ? I would think it would be registered automatically by importing the package, like it is happening with openshift/kubernetes types. Or am I missing something ?

Automatic registration is a bug, not a feature because it assumes a single unified scheme.  As we evolved our API, we developed disjoint schemes focused on their individual purposes.  This improves reliability because it avoids action at a distance.  It is more appropriate to install the types you're using in a local scheme like this.

Reply:
Ok, nice, thanks for the explanation both !